Python/Product/Analysis/Analyzer/InterpreterScope.cs

```diff
@@ -290,8 +290,11 @@ public InterpreterScope(AnalysisValue av, InterpreterScope outerScope)
             _nodeScopes.Clear();
         }
 
-        public virtual IAnalysisSet AddNodeValue(Node node, IAnalysisSet variable) {
-            return _nodeValues[node] = variable;
+        public virtual IAnalysisSet AddNodeValue(Node node, NodeValueKind kind, IAnalysisSet variable) {
+            NodeValue next;
+            _nodeValues.TryGetValue(node, out next);
+            _nodeValues[node] = new NodeValue(kind, variable, next);
```

Comment:
I assume we "never" end up with many entries in here for the same kind, but can we assert that? Could become a memory issue, and certainly could affect Get performance if we end up with 1000s of one kind at the start.

Also, is adding the bottleneck, or is getting? I'd have thought getting, so it may be better to just use a regular list than invent a linked list.